pkg/trigger/discovery_handler.go

```diff
@@ -44,8 +44,11 @@ func (mt *MigrationTrigger) processDiscovery(ctx context.Context) {
 		return true, nil
 	})
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("Abort processing discovery document: %v", err))
-		return
+		if resources == nil {
+			utilruntime.HandleError(fmt.Errorf("Abort processing discovery document: %v", err))
+			return
+		}
+		klog.V(4).Infof("got partial error retrieving discovery document: %v; processing partial result", err)
```

Comment:
`klog.Warningf` is more proper.

Also, I saw the gc code distinguishes if the error is a group discovery error. I don't know why that's needed, but that will add more information during debugging, so let's copy that pattern. See https://github.com/kubernetes/kubernetes/blob/1c11ff7a26c498dc623f060aa30c7c970f3d3eee/pkg/controller/garbagecollector/garbagecollector.go#L640-L646.

Also, can you add a comment stating why we choose to process the partial list?